code/src/Components/MovieDetails.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { SINGLE_MOVIE_URL } from 'utils/urls';
+
+// DISPLAY OF SECOND PAGE AND MORE INFO
+
+const MovieDetails = () => {
+  const [movieDetails, setMovieDetails] = useState([]);
+  const {movieTitle} = useParams();
+  const navigate = useNavigate();
+
+  const onBackButtonClick = () => {
+    // navigate(-1);
+    navigate('/')
+  }
+
+  useEffect(()=> {
+    fetch(SINGLE_MOVIE_URL(movieTitle))
+    .then((res) => res.json())
+    .then((data) => {
+      setMovieDetails(data)
+    })
+  }, [movieTitle]);
```

Comment:
When the first fetch is done using `BASE_URL`, all the details needed are actually provided in that endpoint. So an alternative to doing a new fetch in the details page is to instead take the response from the first fetch and send data as props to `<MovieDetails />` exactly like you do with `<MovieList />`, example:
```javascript
// in Homepage.js

<Route path="/details/:movieTitle" element={<MovieDetails movies={movies}/>} />
```
This might make the loading of the details a bit quicker!